app/controllers/api/v1/users_controller.rb

```diff
@@ -27,6 +27,19 @@ def show
       end
     end
 
+    def search
+      if params[:q].present?
+        users = User.where("name ~* ?", params[:q])
+      else
+        users = User.none
+      end
+      users = users.by_email
```

Comment:
its more a case of having defined behaviour that won't change for no good reason.

I would implement the search action like this:
```
if params[:q].present?
        users = User.where("name ~* ?", params[:q])
      else
        users = User.none
      end
```
as User has a default_scope already ordering by name so there is no need to have the ```by_name``` scope